network/peer/peer.go

```diff
@@ -1099,13 +1099,25 @@ func (p *peer) handlePeerList(msg *p2p.PeerList) {
 			return
 		}
 
+		// Timestamp should be expressible as an int64
+		if claimedIPPort.Timestamp > uint64(math.MaxInt64) {
```

Comment:
Would it make more sense to just have the same:
``` golang
// Note that it is expected that the [ipSigningTime] can be in the past. We
// are just verifying that the claimed signing time isn't too far in the
// future here.
if float64(msg.IpSigningTime)-float64(myTime) > p.MaxClockDifference.Seconds() {
	p.Log.Debug("peer attempting to connect with version timestamp too far in the future",
		zap.Stringer("nodeID", p.id),
		zap.Uint64("ipSigningTime", msg.IpSigningTime),
	)
	p.StartClose()
	return
}
```
check that we do in `Version`?